### PR TITLE
Fixes #11181: Create a technique to download a file from the server

### PR DESCRIPTION
**Pull request by Fdall**
https://www.rudder-project.org/redmine/issues/11181

### CODE REVIEW

**Comment by Fdall** [comment]
Will try to change componentkey to destination file, it is based on the source file which is a non-sense...

File: techniques/fileDistribution/copyGitFile/2.1/copyFileFromSharedFolder.ps1.st [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    for ($i=0; $i -lt $trackingkey.length; $i++) {
+      $local_classes = Merge-ClassContext $local_classes $(File-From-Shared-Folder -Source $names[$i] -Destination $destinations[$i] -HashType $HashCheckType -ReportId $trackingkey[$i] -TechniqueName $techniqueName -auditOnly:$auditOnly -Report:$true -componentName $componentName -componentKey $names[$i])
+      # 8 backslashs are needed because of regex + conversion unix file to windows ? not sure about
+      #this one but this will result in a "-replace '\\\\(\w+)' on windows, as expected
+      $destinations[$i] = $destinations[$i] -replace '\\\\\\\\(\w+)', '\$1'
+      $class_prefix  = "copy_file_" 
+      $class_prefix += Canonify-Class ($names[$i])
+      $class_prefix += "_repaired"
+      $str = $local_classes |out-string
```
Comment (ncharles) [inline techniques/fileDistribution/copyGitFile/2.1/copyFileFromSharedFolder.ps1.st]: where is str used ?

File: techniques/fileDistribution/copyGitFile/2.1/metadata.xml [hunk 2]
```diff
@@ -212,9 +226,11 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
         </CONSTRAINT>
       </INPUT>
 
-    </SECTION>
+  </SECTION>
 
-    <SECTION name="Post-modification hook" component="true" multivalued="false" componentKey="COPYFILE_NAME" displayPriority="low">
+    <SECTION name="WINDOWS specific options" multivalued="false" component="false" displayPriority="low">
```
Comment (ncharles) [inline techniques/fileDistribution/copyGitFile/2.1/metadata.xml]: why is it empty ??

Comment (Fdall) [reply techniques/fileDistribution/copyGitFile/2.1/metadata.xml]: No windows options for the moment, maybe in future versions.

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Need to change the posthook, it does not use GM at the moment

**Comment by Fdall** [comment]
Commit modified

**Comment by amousset** [comment]
This needs to be rebased @Fdall

**Comment by Fdall** [comment]
PR rebased

**Pull request merged**